HSP2tools/readWDM.py

```diff
@@ -7,40 +7,66 @@
 
 import numpy as np
 import pandas as pd
-from numba import jit
+#from numba import jit, njit
 import datetime
+from dateutil.relativedelta import relativedelta
+import timeit
+
+class WDMReader:
+
+    attrinfo = {}
+    freq ={}
+
+    def __init__(self, wdmfile):
+        self.datasets ={}
+        self.dfsummary = pd.DataFrame()
+        self.wdmfile = wdmfile
+
+        # look up attributes NAME, data type (Integer; Real; String) and data length by attribute number
+        self.attrinfo = {1:('TSTYPE','S',4),     2:('STAID','S',16),    11:('DAREA','R',1),
+                17:('TCODE','I',1),     27:('TSBYR','I',1),     28:('TSBMO','I',1),
+                29:('TSBDY','I',1),     30:('TSBHR','I',1),     32:('TFILL', 'R',1),
+                33:('TSSTEP','I',1),    34:('TGROUP','I',1),    45:('STNAM','S',48),
+                83:('COMPFG','I',1),    84:('TSFORM','I',1),    85:('VBTIME','I',1),
+                444:('DATMOD','S',12),  443:('DATCRE','S',12),   22:('DCODE','I',1),
+                10:('DESCRP','S', 80),   7:('ELEV','R',1),       8:('LATDEG','R',1),
+                    9:('LNGDEG','R',1),   288:('SCENARIO','S',8), 289:('CONSTITUENT','S',8),
+                290:('LOCATION','S',8)}
+
+        self.freq = {7:'100YS', 6:'YS', 5:'MS', 4:'D', 3:'H', 2:'min', 1:'S'}   # pandas date_range() frequency by TCODE, TGROUP
+
+
+    def readWDM(self): #, hdffile, compress_output=True):
```

Comment:
I think we could split this main function into a few smaller pieces. I think the first line 39-100 could be some sort of initialize or validate the WDM function. Then the logic in the for loop which loops over the individual DNS records could be it's own function. I think we will ultimately need a way to read a single DSN from a WDM file once we start implementing the IO abstraction class. Additionally that approach would make parallelization easier.